### PR TITLE
allow customized Frapi_Authorization to be injected into the Main controller

### PR DESCRIPTION
**Pull request by clayhinson**
This allows customization of a Frapi_Authorization object before startup, so that the authentication handler can be replaced or extended. 

In my case, this was necessary due to long-running uploads that extend beyond the lifetime of a standard request nonce. I created a custom extension of Frapi_Authorization / Frapi_Authorization_Partner that can override nonce lifetimes by action, and needed a way to inject it. 

If anyone has issues with this, or a better approach let's discuss. (The ability to customize nonce lifetime might be worth adding as a feature itself, as anytime you're handling non-trivial uploads with FRAPI this is going to crop up.)

### CODE REVIEW

File: src/frapi/public/index.php [hunk 1]
```diff
@@ -4,7 +4,10 @@
 set_error_handler(array('Frapi_Error', 'errorHandler'), E_ALL);
 
 try {
-    $controller = new Frapi_Controller_API();
```
Comment (davidcoallier) [inline src/frapi/public/index.php:7]: Would it be possible to add the `{` `}` around the if please?

**Comment by davidcoallier** [comment]
That's a good idea. Just see the comment above and I'll merge it in :)

Thanks!

File: src/frapi/public/index.php (hunk 1 above)
Comment (clayhinson) [inline src/frapi/public/index.php:7]: np, added this and updated the pull request.

**Comment by clayhinson** [comment]
closing, replaced with #123

**Pull request closed without merging**